src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -993,10 +1048,24 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 							if (value is SystemTypeValue systemTypeValue) {
 								foreach (var stringParam in methodParams[1].UniqueValues ()) {
 									if (stringParam is KnownStringValue stringValue) {
-										if (BindingFlagsAreUnsupported (bindingFlags))
+										if (BindingFlagsAreUnsupported (bindingFlags)) {
 											RequireDynamicallyAccessedMembers (ref reflectionContext, DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods, value, calledMethodDefinition);
-										else
-											MarkMethodsOnTypeHierarchy (ref reflectionContext, systemTypeValue.TypeRepresented, m => m.Name == stringValue.Contents, bindingFlags);
+										} else {
+											bool foundAny = false;
+											foreach (var method in systemTypeValue.TypeRepresented.GetMethodsOnTypeHierarchy (m => m.Name == stringValue.Contents, bindingFlags)) {
+												MarkMethod (ref reflectionContext, method);
+												methodReturnValue = MergePointValue.MergeValues (methodReturnValue, new SystemReflectionMethodBaseValue (method));
```

Comment:
Would it make sense to add this handling to RuntimeReflectionExtensions_GetRuntimeMethod as well?

Reply:
Good point - I completely missed that one. Unified the handling and added some tests.